openff/qcsubmit/results/results.py

```diff
@@ -486,7 +486,48 @@ def create_basic_dataset(
         Returns:
             The created basic dataset.
         """
-        raise NotImplementedError()
+
+        records = self.to_records()
+
+        dataset = BasicDataset(
+            dataset_name=dataset_name,
+            description=description,
+            dataset_tagline=tagline,
+            driver=driver,
+            metadata={} if metadata is None else metadata,
+        )
+
+        qc_specs = set()
+
+        for record, molecule in records:
+
+            qc_specs.add(
+                (record.qc_spec.method, record.qc_spec.basis, record.qc_spec.program)
+            )
+
+            dataset.add_molecule(
+                index=molecule.to_smiles(
+                    isomeric=True, explicit_hydrogens=False, mapped=False
+                ),
+                molecule=molecule,
+                attributes=MoleculeAttributes.from_openff_molecule(molecule),
+                extras=record.extras,
+                keywords=record.keywords,
+            )
+
+        dataset.clear_qcspecs()
+
+        for i, (method, basis, program) in enumerate(qc_specs):
```

Comment:
I wonder if it would be better to make a dataset with no `qc_specs` this way the user has to decide or they will get an error at submission, this helps with the awkward case where a user has combined two optimisation datasets with different specifications which will result in the full hessian dataset being run with both which might not be expected?

Reply:
Yeah that's a really good point and I definitely agree - I think in general it probably makes sense for these methods to only populate the entries of the dataset and nothing else (including the metadata) for the reasons you state.

If they do want to just use the same specs from the 'original' dataset(s) then that should only be 1-2 lines extra that they'd need to add to their scripts I think!

Reply:
I partially made this change in `1fe1b32` - I wasn't sure whether we should add no computes specs to the dataset or the default if the user hasn't specified?

Reply:
Ah good point, I think for our use case the default spec should be fine.